core/Plugin/Controller.php

```diff
@@ -395,6 +395,9 @@ protected function getLastUnitGraphAcrossPlugins($currentModuleName, $currentCon
         $date = Common::getRequestVar('date');
         $meta = \Piwik\Plugins\API\API::getInstance()->getReportMetadata($idSite, $period, $date);
 
+        // remove columns from $columnsToDisplay that are not in $selectableColumns
+        $columnsToDisplay = array_intersect($columnsToDisplay, $selectableColumns);
```

Comment:
We'll need to test this one with some other plugins later and in core. I'm not sure if selectable columns and columnsToDisplay is always correctly set at this stage.  This might be better done in Evolution visualization before render when we actually have detected the columnsToDisplay and selectable columns. Reports, Plugins and Visualizations can change both  columns / properties while rendering the visualization